sdks/python/apache_beam/io/gcsio.py

```diff
@@ -47,7 +47,7 @@
       'Google Cloud Storage I/O not supported for this execution environment '
       '(could not import storage API client).')
 
-DEFAULT_READ_BUFFER_SIZE = 1024 * 1024
+DEFAULT_READ_BUFFER_SIZE = 16 * 1024 * 1024
```

Comment:
If you make this buffer size unlimited (e.g. make it 1000 MB), you can tease out a 1.5 to 2x throughput gain (from my local machine), which is not worth the extra memory usage:
> Read 50 MB in 1.43990206718 seconds (34.724583803 MB/s)
> Read 100 MB in 1.40579104424 seconds (71.1343271179 MB/s)
> Read 200 MB in 2.52755904198 seconds (79.1277262681 MB/s)
> Read 400 MB in 4.68462920189 seconds (85.3856266444 MB/s)